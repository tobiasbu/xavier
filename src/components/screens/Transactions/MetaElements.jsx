import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Text from '../../basic/Text';
import useStyle from './style';

/**
 * Total balance
 */
const Balance = (props) => {
  const { totalCurrency } = props;
  const classes = useStyle();
  return (
    <Text size={1} className={classes.fancyBalance}>
      {
        ['Saldo total: ',
          <strong key="totalValue">{totalCurrency}</strong>]
      }
    </Text>
  );
};

/**
 * Balance prop types
 */
Balance.propTypes = {
  totalCurrency: PropTypes.string.isRequired,
};

/**
 * Operation information
 */
const OperationInfo = (props) => {
  const { length, totalCredit, totalDebit } = props;
  const transactionWord = `transaç${(length > 1) ? 'ões' : 'ão'}`;
  return (
    <Text size={3}>
      {
        [
          `Você já realizou o total de ${length} ${transactionWord}.`,
          <br key="really?" />,
          'Valor em operações no Crédito: \n',
          <strong key="totalCredit">{totalCredit}</strong>,
          <br key="really2" />,
          <p style={{ color: 'red' }}>Valor em operações no Débito: </p>, // TODO! Please use astro colors
          <strong key="totalDebit">{totalDebit}</strong>,
        ]
      }
    </Text>
  );
};

/**
 * Operation prop types
 */
OperationInfo.propTypes = {
  length: PropTypes.number.isRequired,
  totalCredit: PropTypes.string.isRequired,
  totalDebit: PropTypes.string.isRequired,
};

/**
 * No register information.
 *
 * The table is empty.
 */
const NoRegister = () => (
  <>
    <p key="p1">
      {
        React.Children.toArray(['Você ainda não cadastrou ',
          <strong>
            nenhuma transação
          </strong>,
          '.'])
      }
    </p>
    <p key="p2">
      {
        React.Children.toArray([
          'Clique na opção ',
          <Link to="/create-transaction" aria-label="Ir para página Nova Transação">Nova Transação</Link>,
          ' e comece a registrar!',
        ])
      }
    </p>
  </>
);

export {
  OperationInfo,
  Balance,
  NoRegister,
};
