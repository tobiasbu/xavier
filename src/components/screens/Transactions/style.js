import { createUseStyles } from 'react-jss';

const useStyle = createUseStyles({
  infoBox: {
    margin: '0 0 2.5rem',
  },
  fancyBalance: {
    borderRadius: '3px',
    background: 'var(--gradient-mayall)',
    padding: '1rem',
    color: 'white !important',
  },
});

export default useStyle;
