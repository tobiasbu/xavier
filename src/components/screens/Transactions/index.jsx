import React from 'react';
import { store } from 'react-notifications-component';

import * as stringUtils from '@utils/stringUtils';
import * as API from '@API';

import Content from '../../container/Content';
import Header from '../../container/Header';

import TransactionTable from '../../compound/TransactionTable';

import { Balance, OperationInfo, NoRegister } from './MetaElements';
import useStyle from './style';

/**
 * Transactions screen.
 */
const Transactions = () => {
  const [transactions, setTransactions] = React.useState(API.getTransactions().reverse());
  const classes = useStyle();

  const updateMyData = (trans) => {
    setTransactions(API.getTransactions().reverse());
    if (trans) { // Fancy
      store.addNotification({
        title: 'Sucesso!',
        message: `Transação "${trans.description}" foi removido com sucesso.`,
        type: 'success',
        container: 'top-right',
        animationIn: ['animated', 'bounceInRight'],
        animationOut: ['animated', 'bounceOut'],
        dismiss: {
          duration: 3500,
          onScreen: true,
        },
        slidingEnter: {
          duration: 750,
        },
        slidingExit: {
          duration: 1000,
        },
      });
    }
  };

  const totalTrans = API.getTotalValueByType();
  const totalCurrency = stringUtils.toCurrency(totalTrans.total);
  const totalDebit = stringUtils.toCurrency(totalTrans.debit);
  const totalCredit = stringUtils.toCurrency(totalTrans.credit);
  const len = transactions.length;

  return (
    <Content>
      <Header>Transações</Header>
      {len > 0
        ? (
          <div className={classes.infoBox}>
            <Balance totalCurrency={totalCurrency} />
            <OperationInfo totalCredit={totalCredit} totalDebit={totalDebit} length={len} />
          </div>
        ) : (
          <NoRegister />
        )}
      <TransactionTable
        transactions={transactions}
        updateTransactions={updateMyData}
      />
    </Content>
  );
};

export default Transactions;
