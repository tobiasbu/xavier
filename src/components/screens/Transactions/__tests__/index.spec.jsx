import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Transactions from '..';

describe('Transactions screen', () => {
  it('should render', () => {
    const component = shallow(
      <Transactions />,
    );
    expect(toJSON(component)).toMatchSnapshot();
  });
});
