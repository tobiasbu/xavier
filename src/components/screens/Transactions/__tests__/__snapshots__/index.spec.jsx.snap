// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions screen should render 1`] = `
<Content>
  <Header>
    Transações
  </Header>
  <NoRegister />
  <TransactionTable
    transactions={Array []}
    updateTransactions={[Function]}
  />
</Content>
`;
